src/components/Checkbox/Checkbox.tsx

```diff
@@ -0,0 +1,97 @@
+import React, { useState } from 'react';
+
+import * as RadixCheckbox from '@radix-ui/react-checkbox';
+import { IconCheckBoxCheck } from '../Icons';
+
+import classNames from 'classnames';
+import styles from './Checkbox.module.scss';
+const cx = classNames.bind(styles);
+
+export interface CheckboxProps {
+  className?: string;
+  name: string;
+  text?: string;
+  link?: string;
+  isDisabled?: boolean;
+  checked?: boolean;
+  hasLightBackground?: boolean;
+  onChange?: (value: boolean) => void;
+  onLinkClick?: () => void;
+}
+
+export const Checkbox = ({
+  className,
+  name,
+  text,
+  link,
+  isDisabled,
+  checked,
+  hasLightBackground,
+  onChange,
+  onLinkClick
+}: CheckboxProps) => {
+  const [isChecked, setIsChecked] = useState(checked);
+
+  const handleChange = (checkState: boolean) => {
+    if (!isDisabled) {
+      setIsChecked(checkState);
+      onChange(checkState);
+    }
+  };
+
+  const handleClick = (event: React.MouseEvent<HTMLSpanElement>) => {
+    if (onLinkClick && !isDisabled) {
+      event.preventDefault();
+      onLinkClick();
+    }
+  };
+
+  const getLabelComponent = () => {
+    if (text && link) {
+      return (
+        <label className={styles.Label} htmlFor={name}>
+          <span>{text}</span>
+          <span onClick={handleClick} className={cx(styles.Link, styles.HasLabel)}>
+            {link}
+          </span>
+        </label>
+      );
+    } else if (link) {
+      return (
+        <label className={cx(styles.Label, styles.HasOnlyLink)} htmlFor={name}>
+          <span onClick={handleClick} className={styles.Link}>
+            {link}
+          </span>
+        </label>
+      );
+    } else {
+      return (
+        <label className={styles.Label} htmlFor={name}>
+          {text}
+        </label>
+      );
+    }
+  };
```

Comment:
Move this to a subcomponent at the bottom of the file. This will make checkbox more readable and maintainable.

```ts
/*************
 * Label
 ************/

interface LabelProps {
    text?: string;
    label?: string;
}

const Label = ({ text, link }) => {
    return (
        <label className={styles.Label} htmlFor={name}>
            {text && ...}
            {link && ...}
        </label>
    )
}
```

You should be able to handle the `HasLabel` and `HasOnlyLink` cases in pure CSS, or perhaps add an attribute to the label, e.g. `<label data-has-label={...} data-has-link={...}`.

Reply:
Updated label as a sub-component